### DISCUSSION TITLE
How does Hexo-server work?

### DISCUSSION THREAD

**Discussion by StackChan** [hexojs/hexo#5079] [Question / Answer] [answered]
If open "index.html" directly in the browser,the website can be shown fully(like some components are not rendered and the page is uncomplete),even running in Idea and running on certain port. But with running "npx hexo server",a website can be fully rendered.
when "index.html" run in Idea or open directly on the browser:
![image](https://user-images.githubusercontent.com/70527681/194048302-33d6f07b-3826-4e36-9f8c-f29371edf365.png)
when "index.html" run with "npx hexo server":
![image](https://user-images.githubusercontent.com/70527681/194048310-b1df3cda-23fb-4c60-a3d5-2dab4520df2e.png)

So,my question is what "npx hexo server" done to make the files in "Public" to run like on the server,instead of on the local computer.

**Comment by stevenjoezhang** [comment, marked as answer]
You have to change root of the http server to the path of `public` folder. For example, run
```
cd test/public
python3 -m http.server
```
